src/pages/quiz.tsx

```diff
@@ -2,79 +2,96 @@ import React, { useState, useEffect } from 'react';
 import quizData from '../quizData.json';
 import { useNavigate } from 'react-router-dom';
 
+type Choice = {
+  value: string;
+  type: string; // e.g., "MCQ"
+  score: number;
+};
+
 type Question = {
   id: number;
   question: string;
-  options: string[];
-  answer: string;
+  options: Choice[]; // use the Choice type here
   reason: string;
-  name: string; // this is not included in the json file
+  name?: string; // you can set this manually or derive it elsewhere
 };
 
 const Quiz: React.FC = () => {
+
   const [currentQuestion, setCurrentQuestion] = useState(0);
-  const [answers, setAnswers] = useState<{ [key: number]: string }>({});
-  const [shuffledOptions, setShuffledOptions] = useState<string[]>([]);
+  const [answers, setAnswers] = useState<{ [key: number]: Choice }>({});
+  const [shuffledOptions, setShuffledOptions] = useState<Choice[]>([]);
 
   const question: Question = quizData[currentQuestion];
   const navigate = useNavigate();
 
+  // shuffed question
   useEffect(() => {
     // Shuffle choices only when the question changes
     setShuffledOptions([...question.options].sort(() => Math.random() - 0.5));
   }, [currentQuestion]);
 
-  const handleAnswer = (answer: string) => {
+  useEffect(() => {
+    console.log("Answers state updated:", answers);
+  }, [answers]);
+
+
+  // answer handler
+  const handleAnswer = (selectedOption: Choice) => {
     setAnswers((prev) => ({
       ...prev,
-      [currentQuestion]: answer, // Store answer for the current question
+      [currentQuestion]: selectedOption, // Store the full choice object
     }));
-
-    // If the answer is correct, automatically go to the next question
-    if (answer === question.answer) {
-      setTimeout(() => {
-        setCurrentQuestion((prev) => (prev + 1) % quizData.length);
-      }, 100); // Delay before moving to next question
-    }
-    
   };
 
+  // what is this?
   const nextQuestion = () => {
-    const is_ended = currentQuestion+1 == quizData.length
-    // console.log(currentQuestion)
-    if(is_ended){
-      navigate('/summary');
-    } else
-    {
+    const is_ended = currentQuestion + 1 === quizData.length;
+
+    if (is_ended) {
+      // Convert answers object to an array if needed
+      const userAnswers = Object.entries(answers).map(([questionIndex, choice]) => ({
+        questionIndex: parseInt(questionIndex),
+        choice
+      }));
+
+      navigate('/summary', { state: { answerLogs: Object.values(answers) } });
+
+    } else {
       setCurrentQuestion((prev) => (prev + 1) % quizData.length);
     }
   };
 
+
   const prevQuestion = () => {
     setCurrentQuestion((prev) => (prev - 1 + quizData.length) % quizData.length);
```

Comment:
may be add stack object storing qurestion ID to pop when previous question is called